player/player.go

```diff
@@ -396,23 +396,33 @@ func (p *Player) PlayPause() error {
 		}
 		return nil
 	case Playing:
-		err := p.setPaused(true)
-		if err == nil {
-			p.prePausedState = p.status.State
-			p.setState(Paused)
-		}
-		return err
+		return p.Pause()
 	case Paused:
-		err := p.setPaused(false)
-		if err == nil {
-			p.setState(p.prePausedState)
-		}
-		return err
+		return p.Continue()
 	default:
 		return errors.New("Unknown player state")
 	}
 }
 
+// Pause playback and update the player state
+func (p *Player) Pause() error {
```

Comment:
maybe add a check: 
```go
if p.status.State != Playing {
    return nil
}
```

since pausing is a no-op if the player isn't playing

Reply:
Good idea, added